uint16 top_left 
uint16 top_right
uint16 bottom_right
uint16 bottom_left